content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md

```diff
@@ -0,0 +1,80 @@
+draft: false
+images: []
+type: docs
+menu:
+  smart-contracts:
+    parent: "smart-contracts-fundamentals"
+    identifier: "aggregators"
+weight: 192
+toc: true
+---
+
+Aggregation is the process of combining multiple packages of _sub-piece data_ into a single package. This large package is stored on the Filecoin network instead of multiple smaller packages. Aggregation can be done off-chain or on-chain.
+
+## Process
+
+The base interface for aggregation requires the following components:
+
+1. A client who has data to upload.
+1. An aggregator platform that clients can interact with to make a deal and retrieve proof of storage.
+1. **Off-chain**: an aggregation node to aggregate the sub-piece data into a larger file
+
+    **On-chain**: an aggregation smart contract that clients can interact with to request to make a deal and verify proof of storage.
+
+![](aggregation-generic.png)
+
+## Proof of Deal Sub-piece Inclusion (PoDSI)
+
+Proof of Deal Sub-piece Inclusion (PoDSI) is motivated by a need for sub-piece data uploads to contain verification and proof that the data was included in an associated deal on Filecoin. PoDSI is heavily used in the aggregated deal-making workflow.
+
+PoDSI is a proofing library that enables the generation of a proof for each sub-piece CID (within the large data segment) and stores it in an off-chain database.
+
+The proof consists of two inclusion proofs:
+
+- An inclusion proof of a sub-tree corresponding to the tree of the client’s data. Optionally, this sub-tree can be misaligned with the proving tree.
+- An inclusion proof of the double leaf data segment descriptor within the data segment index.
+
+A sample [proof is available at lighthouse.storage](https://docs.lighthouse.storage/lighthouse-1/filecoin-virtual-machine/podsi-a-simple-overview).
+
+For further technical discussion, see the [Filecoin Improvement Proposal 512](https://github.com/filecoin-project/FIPs/discussions/512).
+
+### On-chain
+
+For on-chain aggregation smart contracts, developers interact with them to submit sub-piece data and verify proof of storage for their data. The contract also allows for verification of the provided PoDSI, which refers to the sub-piece proof.
+
+Here is the process for on-chain aggregation:
+
+1. The client submits a sub-piece CID (CommPc) with metadata (e.g., metadata of the sub-piece and URL to download the CAR file) directly to the aggregation smart contract.
+1. The aggregator watches the aggregation contract, and when there are enough pieces to produce a 32GiB aggregated piece CID (CommPa), it downloads all sub-pieces for the aggregated piece from the CAR file URL.
+1. The aggregator aggregates indexed data segments into a larger data file for deal-making (based on specs [here](https://github.com/filecoin-project/FIPs/discussions/512)).
+1. The aggregator combines the sub-piece CIDs into the aggregated 32GiB CommP (CommPa) by computing within the aggregation smart contract.
```

Comment:
The commP computation does not happen on chain.

Reply:
This is incorrect:
- Aggregated deal does not have to be 32GiB.
- Subpiece CIDs are not combined into a commP (it's the data).